tls/s2n_server_cert_request.c

```diff
@@ -174,10 +175,8 @@ int s2n_cert_req_send(struct s2n_connection *conn)
         POSIX_GUARD_RESULT(s2n_signature_algorithms_supported_list_send(conn, out));
     }
 
-    /* RFC 5246 7.4.4 - If the certificate_authorities list is empty, then the
-     * client MAY send any certificate of the appropriate ClientCertificateType */
-    uint16_t acceptable_cert_authorities_len = 0;
-    POSIX_GUARD(s2n_stuffer_write_uint16(out, acceptable_cert_authorities_len));
```

Comment:
I think you're getting the two mechanisms mixed up.
* In TLS1.3, we send the extension.
* In TLS1.2, we don't send the extension. `s2n_extension_send_if_tls13_connection` is irrelevant because there's no extension involved. Instead, certificate_authorities is part of the actual CertificateRequest message. That's also why we were writing "0" before-- that's a zero-length certificate_authorities. You can't just write nothing because then you're missing a field and the message is malformed.

Reply:
I was missing the fact that `s2n_cert_req_send` is not called from the tls13 state machine.